phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -550,6 +550,76 @@ public BaseResultIterators(QueryPlan plan, Integer perScanLimit, Integer offset,
         }
         return ranges;
     }
+
+    private List<KeyRange> getRegionRowKeyRanges() throws SQLException {
+        List<HRegionLocation> regionLocations = getRegionBoundaries(scanGrouper);  //Load the region information
+
+        List<KeyRange> regionKeyRanges = Lists.newArrayListWithExpectedSize(regionLocations.size());
```

Comment:
I think we should use Lists.newArrayListWithCapacity(regionLocations.size()), because we know the exact size here. newArrayListWithExpectedSize actually use "given expectsize + 10% more" which we don't need.